### PR TITLE
BGS veteran info caching

### PR DESCRIPTION
**Pull request by pkarman**
connects #1226 

## Description

Caching BGS Veteran info responses per-user to optimize performance.

### CODE REVIEW

File: app/services/external_api/bgs_service.rb [hunk 1]
```diff
@@ -239,4 +239,18 @@ def fetch_user_info(username, station_id = nil, application = "CASEFLOW")
     fail BGS::NoCaseflowAccess if error.message =~ /TODO unknown error string/
     {}
   end
+
+  def bust_fetch_veteran_info_cache(file_number)
```
Comment (tomas-nava) [inline app/services/external_api/bgs_service.rb:243]: Is this only used by tests?

File: app/services/external_api/bgs_service.rb [hunk 2]
```diff
@@ -54,13 +54,15 @@ def parse_veteran_info(veteran_data)
   end
 
   def fetch_veteran_info(file_number, parsed: true)
-    veteran_data =
+    veteran_data = Rails.cache.fetch(fetch_veteran_info_cache_key(file_number), expires_in: 10.minutes) do
```
Comment (tomas-nava) [inline app/services/external_api/bgs_service.rb:57]: Can you say more about the reasoning behind the 10 minute expiration?

File: app/services/external_api/bgs_service.rb (hunk 1 above)
Comment (pkarman) [reply app/services/external_api/bgs_service.rb:243]: yes

File: app/services/external_api/bgs_service.rb (hunk 2 above)
Comment (pkarman) [reply app/services/external_api/bgs_service.rb:57]: Most jobs to download all finish w/in 10 minutes so I wanted to make it long enough to make a difference and not so long that any upstream changes were not reflected "immediately".

**Pull request merged**